lib/index.js

```diff
@@ -824,6 +830,47 @@ class Hatch {
     get_name() { return this._name; }
     get_path() { return this._path; }
 
+    _purge_failed_assets() {
+        // Cache a list of failed asset ids for later
+        const failed_asset_ids = this._failed_assets.map(asset => asset.asset_id);
+
+        let dependents_to_prune = [];
+        let successful_assets = this._assets.filter(asset => {
+            // If an asset failed, dump it from the hatch
+            if (this._failed_assets.includes(asset)) {
+                return false;
+            }
+
+            // If an asset has failed dependents, consider it failed as well
+            const dependents = asset.get_dependent_assets();
+            const has_failed_dependents = dependents.some(dependent_asset_id => {
+                return failed_asset_ids.includes(dependent_asset_id);
+            });
+            if (has_failed_dependents) {
+                // keep track of which siblings we'll have to remove in a later
+                // pass
+                dependents_to_prune = dependents_to_prune.concat(dependents);
+                return false;
+            }
+
+            return true;
+        });
+
+        // In a second pass, remove all otherwise successful assets which had
+        // failed siblings
+        successful_assets = successful_assets.filter(asset => {
+            return !dependents_to_prune.includes(asset.asset_id);
+        });
+
+        // If this hatch had 100% failures, consider it failed and refuse to
+        // continue
+        if (successful_assets.length === 0 && this._assets.length > 0) {
```

Comment:
Maybe this can be set from the caller rather than assuming that any % of failures is fine?

Reply:
maybe if that turns out to be necessary, but tbh I don't really see a use case for it

Reply:
I think there's somewhat of an issue if 99% of articles fail - maybe we can even hard-code some sensible level here?